src/test/java/org/traccar/protocol/HuabaoProtocolDecoderTest.java

```diff
@@ -227,6 +227,22 @@ public void testDecode() throws Exception {
         verifyNull(decoder, binary(
                 "7e0002000004304832546500b7ca7e"));
 
+        verifyAttribute(decoder, binary(
+                "7E020000964130564801050048000000001000004F01651E2E02CAEC3802E80000001524062616002101040000000330011F31010EF1040000308CF231414F56585F564C3330302D4C415F48322E315F4547393135554C4141425230334130324D30385F56322E312E305F763130F3011FF400F5080000000000000001F6084008FFFFFFFD0009F70600000E870246F912000F000000010000062A00082406261600184D7E"),
+                Position.KEY_BATTERY, 3.719);
+
+        verifyAttribute(decoder, binary(
+                "7E020000964130564801050048000000001000004F01651E2E02CAEC3802E80000001524062616002101040000000330011F31010EF1040000308CF231414F56585F564C3330302D4C415F48322E315F4547393135554C4141425230334130324D30385F56322E312E305F763130F3011FF400F5080000000000000001F6084008FFFFFFFD0009F70600000E870246F912000F000000010000062A00082406261600184D7E"),
+                Position.KEY_CHARGE, true);
+
+        verifyAttribute(decoder, binary(
+                "7E020000964130564801050048000000001000004F01651E2E02CAEC3802E80000001524062616002101040000000330011F31010EF1040000308CF231414F56585F564C3330302D4C415F48322E315F4547393135554C4141425230334130324D30385F56322E312E305F763130F3011FF400F5080000000000000001F6084008FFFFFFFD0009F70600000E870246F912000F000000010000062A00082406261600184D7E"),
+                Position.KEY_BATTERY_LEVEL, 70);
+
+        verifyAttribute(decoder, binary(
+                "7E020000964130564801050048000000001000004F01651E2E02CAEC3802E80000001524062616002101040000000330011F31010EF1040000308CF231414F56585F564C3330302D4C415F48322E315F4547393135554C4141425230334130324D30385F56322E312E305F763130F3011FF400F5080000000000000001F6084008FFFFFFFD0009F70600000E870246F912000F000000010000062A00082406261600184D7E"),
+                Position.KEY_POWER, 12.428);
+
```

Comment:
I don't think we need so many test cases. Two is probably enough. One to test `0xF1` and one to test last parameter from `0xF7`.